### PR TITLE
added flags

### PR DESCRIPTION
**Pull request by happyRip**
I added flag functionality to the program. Basically it allows you to modify input without the need to do it manually. Flags have their descriptions, long and short form.

Basic usage:
* Flags have to be given before the rest of the arguments, which become tail, or they wont work.
    * `typer -l 50 file1.txt -r`
this example would not work, because `-r` is given after filename and would be included in `os.Args()` instead of being considered as flag

* `typer -l 50 -r file1.txt file2.txt`
would randomize the imput from both files and limit words to 50

* `typer -shf file1.txt -shf file2.txt file3.txt`
would shuffle the words only in files 1 and 2 (separately) and add file3 in the end.

It is not 100% tested yet, but for now `$(curl ...)` isn't supported yet (need to read files or read text from terminal if an argument is not a path).

Please let me know if you would even consider a functionality like this for me to work further on it.
Also, please give me feedback as it is my first pull request, I hope I did well.

Cheers

### CODE REVIEW

**Comment by maaslalani** [comment]
Hey, thanks so much for opening this PR! I will take a look soon. Just off the bat though, I think we should use a golang library like https://www.github.com/spf13/cobra to do our flag parsing instead of doing it ourselves. This is a fantastic start! I can for sure help you and give feedback if needed 🙂

**Comment by happyRip** [comment]
I've seen the Cobra too and was thinking about it, but went with the `flag` package for now. 
Will look into cobra more and try to work it out, maybe today or tomorrow.

**Comment by maaslalani** [comment]
I had a look over this and I understand the benefits of randomizing the contents of a file and shuffling and limiting but I think we should adhere to the unix philosophy here and simply accept text that the user wants to type. The shuffling and limiting can be all done to the text before passing the input to `typer` for example. `cat README.md | shuf | head -n 10` will shuffle around the README.md contents and take the first ten lines. We can also remove punctuation with `sed` and do more text processing with other unix tools.

So, personally, I think instead of doing this within `typer` we put it on the user to shuffle and randomize and limit their text beforehand.

The two things that I would still like to do are:
- [ ] Use the cobra library
- [ ] Accept text on `stdin` (I haven't figured this out yet still)

**Comment by happyRip** [comment]
With the idea of randomizing I was mostly thinking of dictionaries or other sources of this sort. Generating file before each launch of the application may seem tedious, which made me think to include such tools in the project.

Example given you take the words from [here](https://www.ef.com/wwen/english-resources/english-vocabulary/top-3000-words/). It's 3000 of them. You could either manually generate a file for each launch of the `typer` and manually choose which file to take as input with each launch, or you could set yourself the `typer` with a couple flags and each time get a different input to practice instead of just trying your score.

There is a chance the same can be achieved with `babbler`, but because of some reason it did not work for me and you don't have control over the words list.

I still think using flags may be beneficial to the app, but I guess you don't want your application to become bloated too.

tl;dr: 
Flags still seem to be beneficial to me in some use cases. I'll look into `Cobra` flags, they should be more better, but probably would need an opinion what kind of flags would be a good idea to implement.

**Comment by maaslalani** [comment]
I took a look at https://www.keybr.com/ and saw some interesting stuff, I could see `typer` following a similar style. I'm not against having flags that control the typing test. I just want to make sure they are implemented thoughtfully.

I am trying to follow (copy) keybr.com style.

Let me know what you think of the following:
We introduce two modes:
1. Random Generation (by default)
* Length flag (`--length` / `-l`) integer flag for the number of random words
* Capital flag (`--capital` / `-c`) boolean flag for capital letters (all lower case by default)
* Punctuation (`--punctuation` / `-p`) boolean flag for punctuation (no punctuation by default)
2. Provided Text (specified by the `--file` / `-f` flag to read from a file)
* We introduce a `-f` / `--file` string flag to get input from a file
* Punctuation (`--punctuation` / `-p`) flag (default to `true`, if set to `false` we remove all punctuation from the file). Alternatively, could be `--remove-punctuation`)
* Capital flag (`--capital` / `-c`) flag (default to `true` if set to `false` we lowercase the whole file). Alternatively, could be `--lowercase`

I could definitely see this getting confusing though, which is why I am sort of in favour of simply doing all the text manipulation before hand with the already established unix tools and simply passing them in with pipes to `typer`

E.g. `punctuation` can be removed with `sed` and lowercasing can be done with `tr '[:upper:]' '[:lower:]'`, etc...
